enhancements/node/lazy-image-pull.md

```diff
@@ -0,0 +1,248 @@
+* **Potential Benefits for Large Images:** If the container image is shared between various containers which utilize only part of the image during their execution, Lazy image pulling speeds up such container startup by downloading only the relavent bits from those images.
+
+### User Stories
+
+
+* As a cluster administrator, I want to serve serverless endpoints on my cluster as fast as possible by not waiting for the entire container image to download.
+* As a developer, I want to deploy applications packaged in large container images while minimizing the impact of image size on pod startup time.
+* As a cluster administrator, I want to reduce network bandwidth usage in my OpenShift environment, especially when dealing with large container images.
+
+# Goals
+
+* Modify `c/storage` to support drop-in config
+* Package upstream [Stargz Store plugin](https://github.com/containerd/stargz-snapshotter) for the installation on RHCOS
+* Enable lazy image pulling in Openshift using [Stargz Store plugin](https://github.com/containerd/stargz-snapshotter) which uses [eStargz](https://github.com/containerd/stargz-snapshotter/blob/main/docs/estargz.md) image format.
+* Support lazy image pulling only on the worker nodes (or subset of worker nodes).
+
+# Non-Goals
+
+* Enabling lazy image pulling on the control plane nodes.
+* Replacement of the standard OCI image format; `eStargz` will be an additional supported format.
+* Alterations to the image build process; developers can use standard OCI build tools, such as buildah to build and convert images to `eStargz` format.
+
+## Proposal
+
+In order to use `eStargz` images, we need to have [Stargz Store plugin](https://github.com/containerd/stargz-snapshotter/blob/aaa46a75dd97e401025f82630c9d3d4e41c9f670/docs/INSTALL.md#install-stargz-store-for-cri-opodman-with-systemd) installed on the node and with a configuration for container storage library. Existing MCO [node controller](https://github.com/openshift/machine-config-operator/blob/53bbe70847393935e8f2c3f83c739fbb477cb84d/pkg/controller/node/node_controller.go#L75) for `node.config` object can be used to roll out this configuration to all or [set of specific worker nodes](https://github.com/openshift/machine-config-operator/blob/release-4.16/docs/custom-pools.md).
+
+### Workflow Description
+
+* A cluster administrator will create or edit the [node.config](https://github.com/openshift/api/blob/610cbc77dbab208d5364bba8982385bd148025c1/config/v1/types_node.go#L19) object to enable lazy image pulling support for all worker nodes by, `oc edit nodes.config -o yaml`
+```yaml
+apiVersion: config.openshift.io/v1
+kind: Node
+metadata:
+  name: cluster
+  spec:
+    lazyImagePools:
+      - "worker"
+```
+or for specific machine config pool(s) by,
+```yaml
+apiVersion: config.openshift.io/v1
+kind: Node
+metadata:
+  name: cluster
+  spec:
+    lazyImagePools:
+      - "serverless-custom-pool1"
```

Comment:
I think we should have the type in addition to the pool name. Just in case we want to change the type of the lazy image pull later.